docs/tutorials/vagrant-discovery.rst

```diff
@@ -0,0 +1,376 @@
+
+-  When the API returns "[]"(no nodes discovered so far) RackHD software stacks are running successfully.
+ 
+**2. For demo environment**
+
+- Start RackHD again by typing the following command.
+
+.. code::
+  
+    vagrant@rackhd:~$ sudo service rackhd start
+
+- To use RackHD conveniently, the auth function of RackHD can be closed. Just edit ``/opt/monorail/config.json``, set 16 line `"authEnabled": true` to `"authEnabled": false`. Then restart RackHD service.
+
+.. code::
+
+   vagrant@rackhd:~$ sudo service rackhd restart
+
+-  Wait several seconds, and then type the following RackHD restful API to ensure RackHD is started and is running successfully.
+
+.. code::
+
+    vagrant@rackhd:~$ curl localhost:8080/api/2.0/nodes
+
+-  The following message appears when RackHD is booting and not ready. The process takes several seconds. You can retry after 10 seconds.
+
+.. code::
+
+  curl: (7) Failed to connect to localhost port 8080: Connection refused
+
+-  When the API returns "[]"(no nodes discovered so far) RackHD software stacks are running successfully.
+
+Discovery
+----------
+Vnode is used to do node discovery. A vnode is defined in the file ``Vagrantfile``. The infrasim/quanta_d51 base box is used to set up a vnode. If you want to learn more about Infrasim, you can go to https://github.com/InfraSIM. UltraVNC Viewer is used to show the console of vnode. In this environment, vnode work in the port ``15901``.
+
+1. set up a vnode
+
+``[Note]`` You need to execute below command in the dirctory which contains ``Vagantfile`` on host.If you are in vagrant box, you need to execute ``exit`` command to exit box and then execute below command. 
+
+.. code::
+  
+  sudo vagrant up quanta_d51
+
+You can execute command on host to check whether quanta_d51 vnode is up successfully. If the status of quanta_d51 vnode is ``running``, quanta_d51 is up successfully.
+
+.. code::
+
+  sudo vagrant status
+
+If you are interested in quanta_d51 vnode, you can execute the below command on host to login in quanta_d51 vnode with password: **root**. Then execute the command ``exit`` to enter host.
+
+.. code::
+
+  sudo vagrant ssh quanta_d51
+
+2. On the Windows desktop, double-click the UltraVNC Viewer, and connect to <IP>:15901,to view the PXE progress to boot the microkernel. The <IP> should be set to the node ip where ``quanta_d51 vnode`` is set up by using command **"Vagrant up quanta_d51"**. For example, If you set up environmet on windows, you just need to set <IP> to ``localhost``. If you set up environment on Linux, you just need to get IP by ``ifconfig eth0`` and then replace <IP>.
+
+.. image:: ../_static/node_discovery_1.png
+     :align: center
+
+.. image:: ../_static/node_discovery_2.png
+     :align: center
+
+3. The vNode console pauses for 1 minute, to catalog the node data on this server. After the microkernel boot is completed, the vNode reboots. This reboot indicates that the discovery workflow is completed.
```

Comment:
'about 1 minutes', 'catalog the node information'  what does 'on this server' mean? ‘vNode would  reboot after cataloging finished', it indicates node discovery finished